libs/sdk-ui-filters/src/DateFilter/DateRangePicker/DateTimePicker.tsx

```diff
@@ -0,0 +1,164 @@
+// (C) 2022 GoodData Corporation
+import React from "react";
+import cx from "classnames";
+import { DateRangePickerInputField } from "./DateRangePickerInputField";
+import { injectIntl, WrappedComponentProps } from "react-intl";
+import { IntlWrapper } from "@gooddata/sdk-ui";
+import DayPickerInput from "react-day-picker/DayPickerInput";
+import moment from "moment";
+import { DateRangePickerInputFieldBody } from "./DateRangePickerInputFieldBody";
+import { convertPlatformDateStringToDate } from "../utils/DateConversions";
+import { TIME_FORMAT } from "../constants/Platform";
+
+interface IDateTimePickerOwnProps {
+    placeholderDate: string;
+    dateFormat: string;
+    onChange: (value: Date) => any;
+    value: Date;
+    dayPickerPropsWithDefaults: any;
+    handleDayClick: any;
+    locale: string;
+    isMobile: boolean;
+    isTimeForStaticPeriodEnabled: boolean;
+    className: string;
+    defaultTime?: string;
+    error?: boolean;
+}
+
+export type DateTimePickerComponentProps = IDateTimePickerOwnProps & WrappedComponentProps;
+
+// eslint-disable-next-line react/display-name
+export const DateTimePickerComponent = React.forwardRef<DayPickerInput, DateTimePickerComponentProps>(
+    (props: DateTimePickerComponentProps, ref) => {
+        const {
+            placeholderDate,
+            value,
+            onChange,
+            dateFormat,
+            dayPickerPropsWithDefaults,
+            handleDayClick,
+            isMobile,
+            isTimeForStaticPeriodEnabled,
+            className,
+            defaultTime = "00:00",
+            error = false,
+        } = props;
+
+        const getFormattedDate = () => {
+            return moment(value).format("YYYY-MM-DD");
+        };
+
+        const getFormattedTime = () => {
+            if (typeof value !== ("undefined" || null)) {
+                return moment(value).format(TIME_FORMAT);
+            }
+            return moment(null).format(TIME_FORMAT);
+        };
+
+        const onDateChange = (selectedDate: Date) => {
+            if (isTimeForStaticPeriodEnabled && selectedDate) {
+                let oldDate = value;
+                if (!oldDate) {
+                    oldDate = moment(defaultTime).toDate();
+                }
+
+                let updatedDatetime = moment(selectedDate).hours(oldDate.getHours());
+                updatedDatetime = moment(updatedDatetime).minutes(oldDate.getMinutes());
```

Comment:
I would use two variables and do not override the value. This way you can explain what it does. 

Alternatively, couldn't this be written as 

````
const updatedDateTime = moment(selectedDate)
    .hours(oldDate.getHours())
    .minutes(oldDate.getMinutes());
````
?

Reply:
fixed